### ISSUE TITLE
Bundle.js doesn't properly check for JSON.parse error

### ISSUE THREAD

**Issue by ossgeek314** [CollaboraOnline/online#2289]
**Describe the bug**
When loading a document via nextcloud, bundle.js is loaded and used to process various streams.  The javascript crashes out with a fatal error and depending on when the asynchronous processing happens, the document may fail to load but often just the formatting sidebar remains open.

**To Reproduce**
1. Go to 'any document to be loaded'
2. Click on 'the document'
3. Check javascript console:  `Uncaught SyntaxError: JSON.parse: expected property name or '}' at line 1 column 2 of the JSON data `

**Expected behavior**
The document should load on the webpage and the formatting sidebar should hide itself

**Actual behavior**
The document sometimes doesn't load at all and the formatting sidebar remains visible

**Screenshots**
![Screenshot from 2021-05-06 09-37-46](https://user-images.githubusercontent.com/3986413/117308090-2ddc6500-ae4f-11eb-8ca7-1bfe2613dca9.png)

**Desktop (please complete the following information)**
 - OS: LinuxMint and MacOS tested
 - Browser: Chrome and Firefox tested
 - Version: latest version for each
 - CollaboraOffice 6.4.10.37 (Ubuntu 20.04LTS)
 - loolwsd 6.4.8-4 (Ubuntu 20.04LTS)

**Additional context**
The offending code in bundle.js is:
```javascript
if( typeof e.state == "string" ) {
    var index = e.state.indexOf("{");
    state = index!==-1 ? JSON.parse(e.state.substring(index)) : e.state
}
```
I've been able to prevent the bug by adding a try catch block

```javascript
if( typeof e.state == "string" ) {
    var index = e.state.indexOf("{");
    try{
        state = index!==-1 ? JSON.parse(e.state.substring(index)) : e.state
    } catch {
        state=e.state
    }
}
```

**Comment by ossgeek314** [comment]
It appears to be coming from this file: https://github.com/CollaboraOnline/online/blob/a01e5fb4d88cc5609f7ac59b4d7e3a59b03265f4/loleaflet/src/map/handler/Map.StateChanges.js#L36

**Comment by thebearon** [comment]
This was fixed by 6d63249 in 6.4 and 7a29214 in the main branch.
The fix is part of the 6.4.8-6 release.
